packages/embed-react/src/hooks/use-performance-url.ts

```diff
@@ -1,35 +1,91 @@
-    throw new Error(`WealthSweet context not found and no token was provided to the usePerformanceUrl hook.\n
-      Provide a WealthSweet context with a fetchToken callback for this hook or provide a token in the hook url params`);
+/**
+ * A hook that generates a URL for the WealthSweet performance element.
+ * It uses either provided parameters or context values for token and origin.
+ *
+ * @param {Object} params - The parameters for generating the performance URL.
+ * @param {string} [params.token] - The token to use for authentication.
+ * @param {WealthSweetElementOrigin} [params.origin] - The origin for the WealthSweet element.
+ * @param {...WealthSweetPerforamnceElementQueryParams} params - Additional query parameters for the performance element.
+ *
+ * @returns {Object} An object containing the performance URL and loading state information.
+ * @property {boolean} isTokenLoaded - Indicates whether the token is loaded.
+ * @property {string} [performanceUrl] - The generated performance URL.
+ * @property {boolean} [isTokenError] - Indicates if there was an error fetching the token.
+ * @property {Object} [tokenError] - The error object if there was an error fetching the token.
+ * @property {string} [tokenFetchState] - The current state of token fetching.
+ */
+export function usePerformanceUrl({
+  token: paramToken,
+  origin: paramOrigin,
+  ...params
+}: Partial<
+  WealthSweetPerforamnceElementQueryParams & {
+    origin: WealthSweetElementOrigin;
   }
-  const returnToken = contextLoaded ? contextToken : paramToken;
-  if (returnToken === undefined) {
+>) {
+  const [tokenContextLoaded, tokenContext] = useTokenContextWithoutGuarantee();
+  const [originContextLoaded, originContext] =
+    useOriginContextWithoutGuarantee();
+
+  const origin = chooseHookParamElseContextParam(
+    paramOrigin,
+    originContext?.origin,
+    originContextLoaded,
+    buildContextParamsNotFoundError("usePerformanceUrl", ["origin"]),
+  );
+
+  if (paramToken) {
+    // The token param was passed in so use that.
+    return {
+      isTokenLoaded: true as const,
+      performanceUrl: generateWealthSweetElementUrl({
+        origin,
+        path: "embed/pages/performance",
+        params: { ...params, token: paramToken },
+      }),
+    };
+  }
+  // The token param was not passed in so we need the token context
+  if (!tokenContextLoaded) {
+    throw buildContextParamsNotFoundError("usePerformanceUrl", ["token"]);
+  }
+
+  // Report something bad happened if fetching the token had an error
+  if (tokenContext._tag === "error") {
+    return {
+      isTokenError: true as const,
+      tokenError: tokenContext.error,
+      tokenFetchState: tokenContext.tokenFetchState,
+    };
+  }
+
+  // If there still is no token then we probably have just initialised the hook and the request hasnt fired yet
+  if (!tokenContext.token) {
     return {
       isTokenLoaded: false as const,
+      tokenFetchState: tokenContext.tokenFetchState,
```

Comment:
Do we need to explicitly set `isTokenError` here?
```suggestion
      ifTokenError: false as const,
      isTokenLoaded: false as const,
      tokenFetchState: tokenContext.tokenFetchState,
```